neptune/new/attributes/atoms/artifact.py

```diff
@@ -24,14 +24,17 @@
 
 
 class Artifact(Atom):
-    def _assign(self, value: typing.Union[ArtifactVal, str], wait: bool = False):
-        # this function should not be publicly available
+    def assign(self, value: ArtifactVal, wait: bool = False):
+        # this function should be used only with ArtifactVal
         if not isinstance(value, ArtifactVal):
-            value = ArtifactVal(value)
+            raise TypeError("Value of unsupported type {}".format(type(value)))
 
         with self._run.lock():
             self._enqueue_operation(AssignArtifact(self._path, value.hash), wait)
 
+    def fetch(self) -> ArtifactVal:
+        return ArtifactVal(self.fetch_hash())
```

Comment:
Add `hash=self.fetch_hash()`

Reply:
fun fact: because(?) Artifact extends Atom, its signature is `__init__(self, value: Optional[str] = None)`

I didn't want to expose that, honestly -- and on the other hand I don't want to break the Unified Signature (if there is such a thing)

Reply:
ohh ok